h5grove/content.py

```diff
@@ -147,9 +148,21 @@ def metadata(self, depth=None):
             *super().metadata().items(),
         )
 
-    def data(self, selection: Selection = None, flatten: bool = False):
-        """Dataset data. Supports slicing through selection."""
-        result = get_dataset_slice(self._h5py_entity, selection)
+    def data(
+        self,
+        selection: Selection = None,
+        flatten: bool = False,
+        dtype: Optional[str] = "origin",
+    ):
+        """Dataset data.
+
+        :param selection: Slicing information
+        :param flatten: True to flatten the returned array
+        :param dtype: Data type conversion query parameter
+          - `origin` (default): No conversion
+          - `safe`: Convert to a type supported by JS typedarray (https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/TypedArray)
+        """
+        result = convert(get_dataset_slice(self._h5py_entity, selection), dtype)
```

Comment:
In the end it is where it's belong to me since this is the implementation of the endpoint